apiserver/facades/agent/uniter/networkinfocaas.go

```diff
@@ -164,14 +172,9 @@ func (n *NetworkInfoBase) NetworksForRelation(
 	}
 
 	if len(ingress) == 0 {
-		addrs, err := n.unit.AllAddresses()
-		if err != nil {
-			logger.Warningf("no service address for unit %q in relation %q", n.unit.Name(), rel)
```

Comment:
This is not correct. `AllAddresses` will simple return an empty result if there are none. An error here means something is not right and should be surfaced.